lms/resources/_js_config/__init__.py

```diff
@@ -427,78 +427,40 @@ def _groups(self):
-            sync_api_config["data"]["learner"] = {
-                "canvas_user_id": req.params["learner_canvas_user_id"],
-            }
-
-        return sync_api_config
-
-    def _blackboard_sync_api(self):
-        req = self._request
-        return {
-            "authUrl": req.route_url("blackboard_api.oauth.authorize"),
-            "path": req.route_path("blackboard_api.sync"),
-            "data": {
-                "lms": {
-                    "tool_consumer_instance_guid": self._context.lti_params[
-                        "tool_consumer_instance_guid"
-                    ],
-                    "product": self._request.product.family,
-                },
-                "course": {
-                    "context_id": self._context.lti_params["context_id"],
-                },
-                "assignment": {
-                    "resource_link_id": self._context.lti_params["resource_link_id"],
-                    "group_set_id": self._context.group_set_id,
-                },
-                "group_info": {
-                    key: value
-                    for key, value in self._context.lti_params.items()
-                    if key in GroupInfo.columns()
-                },
-            },
-        }
-
     def _sync_api(self):
         if self._context.grouping_type == Grouping.Type.COURSE:
             return None
 
-        if self._request.product.family == Product.Family.CANVAS:
-            return self._canvas_sync_api()
-
-        if self._request.product.family == Product.Family.BLACKBOARD:
-            return self._blackboard_sync_api()
+        plugin = self._request.product.plugin.grouping_service
+        if not plugin.sync_route:
+            return None
 
-        return None
+        return {
+            "authUrl": self._request.route_url(plugin.auth_route),
+            "path": self._request.route_path(plugin.sync_route),
+            "data": plugin.get_grouping_sync_config(
+                self._request,
+                {
+                    "lms": {
+                        "tool_consumer_instance_guid": self._context.lti_params[
+                            "tool_consumer_instance_guid"
+                        ],
+                        "product": self._request.product.family,
+                    },
+                    "course": {
+                        "context_id": self._context.lti_params["context_id"],
+                    },
+                    "assignment": {
+                        "resource_link_id": self._context.lti_params[
+                            "resource_link_id"
+                        ],
+                        "group_set_id": self._context.group_set_id,
+                    },
+                    "group_info": {
+                        key: value
+                        for key, value in self._context.lti_params.items()
+                        if key in GroupInfo.columns()
+                    },
+                },
+            ),
+        }
```

Comment:
This whole business could be in the default method in the plugin, however it's a bit awkward while we rely on the context object for `group_set_id`.

Also all the `plugin.*` becomes `self.*`.